signalfx.gemspec

```diff
@@ -27,24 +27,17 @@ Gem::Specification.new do |spec|
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ["lib"]
 
-  spec.required_ruby_version = '>= 2.2.0'
+  spec.required_ruby_version = '>= 2.2.2'
 
-  spec.add_development_dependency "bundler", "~> 1.17.3"
+  spec.add_development_dependency "bundler", ">= 1.17.3"
   spec.add_development_dependency "rake", "~> 10.0"
   spec.add_development_dependency "rspec", "~> 3.3"
   spec.add_development_dependency "webmock", "~> 2.3.1"
   spec.add_development_dependency "thin", "~> 1.7"
   spec.add_development_dependency "pry"
 
-  # protobuf enforces this check but builds with a newer Ruby version so it's not enabled.
-  # Incorporating here to allow 2.2.0-1 users to successfully build and install signalfx.
-  active_support_max_version = Gem::Version.new(RUBY_VERSION) < Gem::Version.new("2.2.2") ? "<5" : "<6"
-  spec.add_dependency "activesupport", '>= 3.2', active_support_max_version
```

Comment:
Would prefer we not remove assistance, let alone support, in these cases. removing the cap should help?

Reply:
I understand the intent, but if the same code is already in protobuf, what is changed by adding it to our gemspec?

Reply:
As attempted to be explained in the comment, since this condition is evaluated at build time I don't believe it's of assistance for those using 2.2.(0,1).  I'm not sure why the <6 restriction was added and defer to @keitwb https://github.com/signalfx/signalfx-ruby/pull/43/files#diff-565c020e7394b6c19aa8a85842735375b305a476bc048cae1ec0d7c8ce4ebd66R41

My thought would be to revert to what was originally there (no ceiling) and as documented if not using bundler.

Reply:
Okay I think I now understand what you meant. I agree there's an issue with installing gems using `gem install ...`, be it from a .gem file or from RubyGems.  I understand it's a valid use case. I have looked at other gems to check how they solve the problem, and I think that the general sentiment is to use Bundler, rather than run `gem install` directly, unless it's to install a specific version of a gem.

As a one example, take a look at:
```bash
root@076de655ca44:~/signalfx-ruby# cat /etc/issue
Ubuntu 16.04.7 LTS \n \l

root@076de655ca44:~/signalfx-ruby# ruby -v
ruby 2.2.3p173 (2015-08-18 revision 51636) [x86_64-linux]
root@076de655ca44:~/signalfx-ruby# gem install activesupport
ERROR:  Error installing activesupport:
        i18n requires Ruby version >= 2.3.0.
```
What I take away form this is that even Rails' core team doesn't care about this use case. My intuition here is that `gem install` doesn't resolves gem versions without checking for supported ruby version, and as a result resolves a version which is too high.

I don't think it's reasonable to expect us to solve these issues for our users.

Reply:
The issue had been solved in the same spirit protobuf solves and the actual problem as I see it was the addition of the cap to '<6', which should be removed.  Am I missing something?

Reply:
For older versions of ruby, they need to build the gem themselves as rubygems doesn't provide assistance.  We are trying to help them and lose nothing in the process afaict.